src/components/calcite-dropdown/calcite-dropdown.tsx

```diff
@@ -432,17 +433,24 @@ export class CalciteDropdown {
     const { maxItems } = this;
     let itemsToProcess = 0;
     let maxScrollerHeight = 0;
+    let borderHeight = 0;
 
     groups.forEach((group) => {
       if (maxItems > 0 && itemsToProcess < maxItems) {
-        maxScrollerHeight += group?.titleEl?.offsetHeight || 0;
-
         group.items.forEach((item) => {
           if (itemsToProcess < maxItems) {
             maxScrollerHeight += item.offsetHeight;
             itemsToProcess += 1;
           }
         });
+
+        const separatorEl = group?.group?.shadowRoot.querySelector(
```

Comment:
Is there a way you can get the offset height without querying the group's internals? We should avoid peeking into the internals of other components.

Reply:
Hey, I meant I had to add this to get the test for intersecting viewport to pass when two groups were present (due to the extra px height from the separator), not update the test itself. I could probably update this to instead subtract the offset height of the total items in a group from the group itself... can look

Reply:
@jcfranco just updated to pass up the separator element in the same manner the title element is, can you give this another look?